File: clean_firestore.js
#!/usr/bin/env node
/**
 * Clean Firestore database - Delete all documents from all collections
 */

const admin = require('firebase-admin');

// Initialize Firebase Admin with application default credentials
admin.initializeApp({
  projectId: 'streamsense-476705',
});

const db = admin.firestore();

async function deleteCollection(collectionName, batchSize = 100) {
  console.log(`\n🗑️  Deleting collection: ${collectionName}`);
  
  const collectionRef = db.collection(collectionName);
  let deleted = 0;
  
  while (true) {
    const snapshot = await collectionRef.limit(batchSize).get();
    
    if (snapshot.empty) {
      break;
    }
    
    const batch = db.batch();
    snapshot.docs.forEach((doc) => {
      batch.delete(doc.ref);
      deleted++;
    });
    
    await batch.commit();
    process.stdout.write(`   Deleted ${deleted} documents...\r`);
  }
  
  console.log(`   ✅ Deleted ${deleted} documents from ${collectionName}`);
  return deleted;
}

async function main() {
  console.log('='.repeat(70));
  console.log('🧹 CLEANING FIRESTORE DATABASE');
  console.log('='.repeat(70));
  
  // Collections to clean
  const collections = [
    'messages',           // Chat messages
    'trends',            // Trend analysis
    'insights',          // AI insights
    'agent_activity',    // Agent activity logs
    'users',             // User profiles
  ];
  
  let totalDeleted = 0;
  
  for (const collection of collections) {
    try {
      const deleted = await deleteCollection(collection);
      totalDeleted += deleted;
    } catch (error) {
      console.log(`   ❌ Error deleting ${collection}: ${error.message}`);
    }
  }
  
  console.log('\n' + '='.repeat(70));
  console.log(`✅ CLEANUP COMPLETE - Deleted ${totalDeleted} total documents`);
  console.log('='.repeat(70));
  
  // Keep config/twitch_channel but reset monitoring_enabled
  console.log('\n🔧 Resetting monitoring config...');
  try {
    const configRef = db.collection('config').doc('twitch_channel');
    await configRef.update({
      monitoring_enabled: true,
      updatedAt: admin.firestore.FieldValue.serverTimestamp()
    });
    console.log('   ✅ Reset monitoring_enabled to True');
  } catch (error) {
    console.log(`   ⚠️  Could not reset config: ${error.message}`);
  }
  
  console.log('\n🎉 Database is now clean! You can sign in with a new account.');
  process.exit(0);
}

main().catch((error) => {
  console.error('❌ Error:', error);
  process.exit(1);
});

File: backend/dashboard-api/clean_firestore.go
package main

import (
	"context"
	"fmt"
	"log"

	"cloud.google.com/go/firestore"
	"google.golang.org/api/iterator"
)

func deleteCollection(ctx context.Context, client *firestore.Client, collectionName string, batchSize int) error {
	fmt.Printf("\n🗑️  Deleting collection: %s\n", collectionName)
	
	deleted := 0
	for {
		// Get a batch of documents
		iter := client.Collection(collectionName).Limit(batchSize).Documents(ctx)
		numDeleted := 0

		// Create a batch
		batch := client.Batch()
		for {
			doc, err := iter.Next()
			if err == iterator.Done {
				break
			}
			if err != nil {
				return fmt.Errorf("error iterating documents: %v", err)
			}

			batch.Delete(doc.Ref)
			numDeleted++
		}

		// If there are no documents to delete, we're done
		if numDeleted == 0 {
			break
		}

		// Commit the batch
		_, err := batch.Commit(ctx)
		if err != nil {
			return fmt.Errorf("error committing batch: %v", err)
		}

		deleted += numDeleted
		fmt.Printf("   Deleted %d documents...\r", deleted)
	}

	fmt.Printf("   ✅ Deleted %d documents from %s\n", deleted, collectionName)
	return nil
}

func main() {
	ctx := context.Background()

	// Initialize Firestore client
	projectID := "streamsense-476705"
	client, err := firestore.NewClient(ctx, projectID)
	if err != nil {
		log.Fatalf("Failed to create Firestore client: %v", err)
	}
	defer client.Close()

	fmt.Println("======================================================================")
	fmt.Println("🧹 CLEANING FIRESTORE DATABASE")
	fmt.Println("======================================================================")

	// Collections to clean
	collections := []string{
		"messages",        // Chat messages
		"trends",          // Trend analysis
		"insights",        // AI insights
		"agent_activity",  // Agent activity logs
		"users",           // User profiles
	}

	totalDeleted := 0
	for _, collection := range collections {
		err := deleteCollection(ctx, client, collection, 100)
		if err != nil {
			fmt.Printf("   ❌ Error deleting %s: %v\n", collection, err)
		}
	}

	fmt.Println("\n======================================================================")
	fmt.Printf("✅ CLEANUP COMPLETE - Deleted %d total documents\n", totalDeleted)
	fmt.Println("======================================================================")

	// Reset monitoring config
	fmt.Println("\n🔧 Resetting monitoring config...")
	configRef := client.Collection("config").Doc("twitch_channel")
	_, err = configRef.Update(ctx, []firestore.Update{
		{Path: "monitoring_enabled", Value: true},
		{Path: "updatedAt", Value: firestore.ServerTimestamp},
	})
	if err != nil {
		fmt.Printf("   ⚠️  Could not reset config: %v\n", err)
	} else {
		fmt.Println("   ✅ Reset monitoring_enabled to True")
	}

	fmt.Println("\n🎉 Database is now clean! You can sign in with a new account.")
}

